design/kata-design-requirements.md

```diff
@@ -0,0 +1,95 @@
+following `runc` commands:
+
+* `create`
+* `delete`
+* `exec`
+* `kill`
+* `list`
+* `pause`
+* `ps`
+* `start`
+* `state`
+* `version`
+
+The Kata Containers runtime **MUST** implement the following command line options:
+* `--console-socket`
+* `--pid-file`
+
+### CRI and Kubernetes support
+The Kata Containers project **MUST** provide two interfaces for CRI shims to manage hardware
+virtualization based Kubernetes pods and containers:
+-  An OCI and `runc` compatible command line interface, as decribed in the previous section.
+This interface is used by implementations such as `CRI-O` and `cri-containerd`, for example.
+- A hardware virtualization runtime library API for CRI shims to consume and provide a more
+CRI native implementation, The `frakti` CRI shim is an example of such a consumer.
+
+### Multiple hardware architectures support
+The Kata Containers runtime **MUST NOT** be architecture-specific. It should be able to support
+multiple hardware architectures and provide a modular and flexible design for adding support
+for additional ones.
+
+### Multiple hypervisor support
+The Kata Containers runtime **MUST NOT** be tied to any specific hardware virtualization technology,
+hypervisor, or virtual machine monitor implementation.
+It should support multiple hypervisors and provide a pluggable and flexible design to add support
+for additional ones.
+
+#### Nesting
+The Kata Containers runtime **MUST** support nested virtualization environments.
+
+### Networking
+
+* The Kata Containers runtime **MUST** support any CNI plugin.
+* The Kata Containers runtime **MUST** support both legacy and IPv6 networks.
+
+### I/O
+
+#### Devices direct assignment
+In order for containers to directly consume host hardware resources, the Kata Containers runtime
+**MUST** provide containers with secure pass through for generic devices like e.g. GPUs, SRIOV,
+RDMA, QAT, by leveraging I/O virtualization technologies (IOMMU, interrupt remapping, etc...).
+
+#### Acceleration
+The Kata Containers runtime **MUST** support accelerated and user-space-based I/O operations
+for networking (e.g. DPDK) as well as storage through `vhost-user` sockets.
+
+#### Scalability
+The Kata Containers runtime **MUST** support scalable I/O through the SRIOV technology.
+
+
+### Virtualization overhead reduction
+A compelling aspect of containers is its minimal overhead compared to bare metal applications.
+A container runtime should keep the overhead to a minimum in order to provide the expected user
+experience. 
+The Kata Containers runtime implementation **SHOULD** be optimized for:
+
+* Minimal workload boot and shutdown times
+* Minimal workload memory footprint
+* Maximal networking throughput
+* Minimal networking latency
+
+### Testing and debugging
+
+#### Continuous Integration
+Each Kata Containers runtime pull request **MUST** pass at least the following  set of container-related
+tests:
```

Comment:
I'm not sure if it's appropriate to add the following to this doc, but if you want to be comprehensive the PRs also have to pass the following as part of the CI:

- a whole battery of static analysis and linter tests (including cyclomatic complexity).
- formatting checks.
- distro testing (running the PR on different distro versions).
- log format checking (with the advent of https://github.com/kata-containers/tests/pull/98 this now happens automatically for every PR).
- [`checkcommits`](https://github.com/kata-containers/tests/tree/master/cmd/checkcommits) commit sanity checker.
- PnP tests to ensure the change doesn't regress performance.

That's in addition to the following of course:

- documentation review (where appropriate).
- multiple code inspection reviews (pullapprove acks).

Reply:
I think we should add a document specific to our continuous integration and additional tests which can go into much more detail.

Reply:
Sounds good.